src/actions/upgrade-solution/index.ts

```diff
@@ -1,40 +1,36 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 import * as core from '@actions/core';
-import { AuthHandler, AuthKind, DefaultRunnerFactory, getInputAsBool, RunnerFactory } from '../../lib';
+import { upgradeSolution } from "@microsoft/powerplatform-cli-wrapper/dist/actions";
+import { YamlParser } from '../../lib/parser/YamlParser';
+import { ActionsHost } from '../../lib/host/ActionsHost';
+import getCredentials from "../../lib/auth/getCredentials";
+import getEnvironmentUrl from "../../lib/auth/getEnvironmentUrl";
+import { runnerParameters } from '../../lib/runnerParameters';
 
 (async () => {
     if (process.env.GITHUB_ACTIONS) {
-        await main(DefaultRunnerFactory);
+        await main();
     }
 })();
 
-export async function main(factory: RunnerFactory): Promise<void> {
-    let pac;
+export async function main(): Promise<void> {
     try {
         core.startGroup('upgrade-solution:');
-        const solutionName = core.getInput('solution-name', { required: true });
-        core.info(`solution: ${solutionName}`);
+        const taskParser = new YamlParser();
+        const parameterMap = taskParser.getHostParameterEntries(runnerParameters.workingDir, "upgrade-solution");
 
-        const performAsync = getInputAsBool('async', false, false);
-
-        pac = factory.getRunner('pac', process.cwd());
-        await new AuthHandler(pac).authenticate(AuthKind.CDS);
-        const upgradeArgs = ['solution', 'upgrade', '--solution-name', solutionName];
-        if (performAsync) {
-            upgradeArgs.push('--async');
-
-            const maxAsyncWaitTime = core.getInput('max-async-wait-time', { required: false });
-            if (maxAsyncWaitTime) { upgradeArgs.push('--max-async-wait-time', maxAsyncWaitTime); }
-        }
-
-        await pac.run(upgradeArgs);
-        core.info('upgraded solution');
+        core.info("solution name: " + parameterMap["solution-name"]);
```

Comment:
Is this required, aren't we already logging parameters like solution name in cli-wrapper?

Reply:
Just transferred what was existing before